### PR TITLE
Add parser unit tests

### PR DESCRIPTION
**Pull request by kiwiyou**
- [x] test line 0
- [x] test line 1
- [x] test line 2
- [x] test line 3
- [x] test line 4
- [x] test line 5
- [x] test single document
- [x] test multipart document
- [x] test color definition

### CODE REVIEW

File: ldraw/src/color.rs [hunk 1]
```diff
@@ -8,6 +8,7 @@ use serde::{Deserialize, Serialize};
 
 use crate::Vector4;
 
+#[cfg_attr(test, derive(PartialEq))]
```
Comment (segfault87) [inline ldraw/src/color.rs:11]: I think's it's completely fine to derive `PartialEq`s by default.

File: ldraw/src/parser.rs [hunk 2]
```diff
@@ -747,72 +747,629 @@ pub async fn parse_color_definition<T: BufRead + Unpin>(
...
+0 BFC CERTIFY CCW
+
+
+5 24 0 -55.673 -15.623 0 -59.974 -18.831 4.233 -59.338 -18.968 -4.233 -59.338 -18.968";
+        let parsed = parse_multipart_document(&colors, &mut document.as_bytes())
+            .await
+            .unwrap();
+        let mut subparts = HashMap::new();
+        subparts.insert(
+            PartAlias {
```
Comment (segfault87) [inline ldraw/src/parser.rs:1325]: You can use `PartAlias::from("apple.ldr".into())` instead.

Comment (segfault87) [reply ldraw/src/parser.rs:1325]: Otherwise of that it looks fine. Thanks for your contribution!

Comment (kiwiyou) [reply ldraw/src/parser.rs:1325]: It seems `PartAlias::from("apple.ldr".to_string())` is available. Thanks!

**Review by segfault87 — APPROVED** [review]
:100:

**Pull request merged**